inc/class-wpseo-utils.php

```diff
@@ -195,34 +195,35 @@ public static function trim_recursive( $value ) {
 	 * @return string
 	 */
 	public static function translate_score( $val, $css_value = true ) {
-		if ( $val > 10 ) {
-			$val = round( $val / 10 );
-		}
-		switch ( $val ) {
-			case 0:
+
+		switch ( true ) {
+
+			case 0 === $val:
 				$score = __( 'N/A', 'wordpress-seo' );
-				$css   = 'na';
+				$css   = WPSEO_Rank::NO_FOCUS;
+				break;
+
+			default:
+			case WPSEO_Rank::$range[ WPSEO_Rank::BAD ]['start'] <= $val && $val <= WPSEO_Rank::$range[ WPSEO_Rank::BAD ]['end']:
+				$score = __( 'Bad', 'wordpress-seo' );
+				$css   = WPSEO_Rank::BAD;
 				break;
-			case 4:
-			case 5:
+
+			case WPSEO_Rank::$range[ WPSEO_Rank::POOR ]['start'] <= $val && $val <= WPSEO_Rank::$range[ WPSEO_Rank::POOR ]['end']:
 				$score = __( 'Poor', 'wordpress-seo' );
-				$css   = 'poor';
+				$css   = WPSEO_Rank::POOR;
 				break;
-			case 6:
-			case 7:
+
+			case WPSEO_Rank::$range[ WPSEO_Rank::OK ]['start'] <= $val && $val <= WPSEO_Rank::$range[ WPSEO_Rank::OK ]['end']:
 				$score = __( 'OK', 'wordpress-seo' );
```

Comment:
Instead of having the labels here, could you have them in the `WPSEO_Rank` class?

i.e.
~~~php
'good' => array(
	'start' => 75,
	'end'   => 100,
	'label' => __( 'Good', 'wordpress-seo' ),
),
~~~

That way, all of the configuration is kept together and the label can be more easily used in another place.